fe/fe-core/src/main/java/com/starrocks/catalog/CatalogRecycleBin.java

```diff
@@ -521,6 +521,10 @@ public synchronized boolean recoverTable(Database db, String tableName) {
         }
 
         Table table = recycleTableInfo.getTable();
+        if (idToRecycleTime.get(table.getId()) <= 0) {
```

Comment:
If delete action is executed failed and FE is restarted, recycle time will be restored to the current time.